### ISSUE TITLE
Magento 2 product video is not playing for configurable product

### ISSUE THREAD

**Issue by karnasaradar** [magento/magento2#23107]
I have uploaded a youtube video for a configurable product. But it's showing as a still image , not showing any play icon and it's not playing.

Can anyone give me a solution for the issue?

**Comment by AlexWorking** [comment]
Hello, @karnasaradar. In order to process Your issue, please update Your comment where Your describe the issue with all the required steps as the bot asks. Thank You.

**Comment by codesource** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @codesource. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @codesource, here is your [Magento instance](https://i-23107-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23107-2-3-develop.instances.magento-community.engineering/admin](https://i-23107-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by codesource** [comment]
@magento give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @codesource. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @codesource, here is your [Magento instance](https://i-23107-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-23107-2-2-develop.instances.magento-community.engineering/admin](https://i-23107-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sdzhepa** [comment]
Hello @karnasaradar 

Unfortunately, it is very hard to figure out what issue you faced based on the provided description. It does not have any information about preconditions, steps, version and etc.

Please have a look at [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) from more details.


I tried to investigate and reproduce it but it works for me.
I suppose it could be because of missed or invalid **YouTube API Key** _in Stores > Configuration > CATALOG > Catalog > Product Video_

Please see this video for more details: 
https://www.screencast.com/t/nXn368BMiY8

I cannot confirm the issue and have to close it for now.
Please feel free to provide all required details into the description and reopen the issue

**Comment by codesource** [comment]
For what I have tested, on last version of each branch it works.

@karangar You should try to upgrade your Magento with last version and test it again

**Comment by MagDevX** [comment]
I confirm that this issue exists in Magneto CE 2.3.5-p1, but it is VERY tricky to reproduce! I spent hours comparing two seemingly identical listings to figure why one of them shows the videos and the other one doesn't.

**TL;DR The order of adding videos and images matters. The videos should be uploaded first, then the images should be added.**

In order to reproduce the issue follow these steps:
- Create a configurable product containing 3 simple products that vary in color (make sure to upload an image for each simple product as well as the configurable product, then save the configurable product).
- Now that the listing is published, add a YouTube video to each simple product and move it to the first position (to the left of the image).
- If you check the frontend and switch between the options (i.e. simple products), you will see that the video is just a thumbnail and will not play.

Workaround:
- Remove the images from the simple products, so that only the video remains, then save the simple products.
- Now add the images back to the simple products.
